libs/components/layout/testing/src/modules/text-expand/text-expand-harness.ts

```diff
@@ -0,0 +1,66 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+
+import { SkyTextExpandHarnessFilters } from './text-expand-harness-filters';
+import { SkyTextExpandModalHarness } from './text-expand-modal-harness';
+
+/**
+ * Harness for interacting with a text expand component in tests.
+ */
+export class SkyTextExpandHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-text-expand';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+  #getExpandCollapseButton = this.locatorForOptional(
+    '.sky-text-expand-see-more',
+  );
+  #getText = this.locatorFor('.sky-text-expand-text');
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyTextExpandHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyTextExpandHarnessFilters,
+  ): HarnessPredicate<SkyTextExpandHarness> {
+    return SkyTextExpandHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the button element that expands or collapses text.
+   */
+  public async clickExpandCollapseButton(): Promise<void> {
```

Comment:
The "toggle" action isn't very helpful in a testing scenario, since it tells us nothing about the current state of the component. Could we add a `isExpanded` method that will let the user determine the state of the expansion after clicking the button? Similar to [how we handle popovers](https://github.com/blackbaud/skyux/blob/main/libs/components/popovers/testing/src/modules/popover/popover-harness.ts#L51-L53).
```
await harness.clickExpandCollapseButton();
await expectAsync(harness.isExpanded()).toBeResolvedTo(true);
```